compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaBasedOwner.skiko.kt

```diff
@@ -75,16 +75,13 @@ internal class SkiaBasedOwner(
     initLayoutDirection: LayoutDirection = platform.layoutDirection,
     bounds: IntRect = IntRect.Zero,
     val focusable: Boolean = true,
-    val onClickOutside: (() -> Unit)? = null,
+    val onOutsidePointerEvent: ((PointerInputEvent) -> Unit)? = null,
```

Comment:
Passing `(PointerInputEvent) -> Boolean = { false }` instead will help us to not consume non-left clicks for dialogs:
```
     val onOutsidePointerEvent = { event: PointerInputEvent ->
            if (properties.dismissOnClickOutside && event.isDismissRequest()) {
                onDismissRequest()
                true
            } else {
                false
            }
        }
    }
```

Reply:
It's not relevant here because it's already called after all conditions and if `pressOwner == null`

Reply:
After second look, it is indeed, not needed in the current code, because we don't check if onDismissRequest is null. 

The suggestion was to add `consume` logic.

Imagine hypothetical non-focusable context menu that should be closed on the left click outside, but it should pass the right click to the main owner to reopen the context menu. Just an idea for further improvements in the future, (if it makes sense). The current code is fine.

Reply:
> The suggestion was to add consume logic.

It's all internal, we can add it later once (if) needed.

> non-focusable context menu that should be closed on the left click outside, but it should pass the right click to the main owner to reopen the context menu

This case works now, it doesn't touch consuming of click-outside event. It was mentioned by @m-sasha during discussion and this is why I removed primary button check for popups.
How it works: non-focusable popup doesn't block passing event through, so old context menu will receive onDismiss AND background will receive this gesture. If it's right-click - it will open a new one.

Reply:
> It's all internal, we can add it later once (if) needed.

Sure, just worth to keep in mind